src/lib.rs

```diff
@@ -315,24 +422,35 @@ macro_rules! impl_trait(
                 }
             }
         }
+
+        #[cfg(feature = "alloc")]
+        unsafe impl FromByteVec for $to {
+            fn from_byte_vec(mut vec: Vec<u8>) -> Result<Vec<$to>, Error> {
+                let (len, capacity) = check_vec_constraints::<$to>(&vec)?;
+                let ptr = vec.as_mut_ptr();
+                mem::forget(vec);
+                unsafe {
+                    Ok(Vec::from_raw_parts(ptr as *mut $to, len, capacity))
```

Comment:
You have to special-case empty vecs like here: https://github.com/sdroege/byte-slice-cast/blob/d8ab833a0c0cda6dfe304b183f336514260f0f22/src/lib.rs#L272-L273

if capacity is 0, return `vec![]`

Reply:
And in that case don't forget the input vector :)

Reply:
Ah Okay :-).